src/components/avatar/daikin-avatar.ts

```diff
@@ -0,0 +1,194 @@
+          "[&>span]:disabled:text-ddt-color-common-surface-default",
+        ],
+        icon: [
+          "bg-ddt-color-common-neutral-default",
+          "text-ddt-color-common-surface-default",
+        ],
+        link: [
+          "link-enabled:bg-ddt-color-common-neutral-default",
+          "[&>span]:link-enabled:text-ddt-color-common-surface-default",
+
+          "link-enabled:hover:bg-ddt-color-common-neutral-hover",
+          "[&>span]:link-enabled:hover:text-ddt-color-common-surface-hover",
+          "link-enabled:active:bg-ddt-color-common-neutral-press",
+          "[&>span]:link-enabled:active:text-ddt-color-common-surface-press",
+          "link-disabled:bg-ddt-color-common-disabled",
+          "[&>span]:link-disabled:text-ddt-color-common-surface-default",
+        ],
+      },
+    },
+  }
+);
+
+type AvatarVariantProps = MergeVariantProps<typeof cvaWrapper>;
+
+/**
+ * The avatar is a UI element that represent a user
+ *
+ * @fires click - A retargeted event of a [click event](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) emitted from the inner `<button>` element. Suppressed if `disabled` is true,
+ *
+ * @example
+ *
+ * ```js
+ * import "@daikin-oss/design-system-web-components/components/avatar/index.js";
+ * ```
+ *
+ * ```html
+ * <daikin-avatar></daikin-avatar>
+ * ```
+ */
+@customElement("daikin-avatar")
+export class DaikinAvatar extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-flex;
+      position: relative;
+    }
+
+    :host([size="small"]) {
+      padding: 4px;
+    }
+
+    :host([size="large"]) {
+      padding: 10px;
+    }
+  `;
+
+  /**
+   * Size of the avatar.
+   */
+  @property({ type: String, reflect: true })
+  size: AvatarVariantProps["size"] = "small";
+
+  /**
+   * Replace the wrapping element.
+   */
+  @property({ type: String, reflect: true })
+  as: "icon" | "button" | "link" = "icon";
+
+  /**
+   * Link `href`.
+   * Only used if the `as` is `"link"`.
+   * If omitted with `as="link"`, the link will be treated as [a placeholder link](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element:~:text=If%20the%20a%20element%20has%20no%20href%20attribute) and rendered as disabled state.
+   */
+  @property({ type: String, reflect: true })
+  href: string | null = null;
+
+  /**
+   * Provides an accessible name of the avatar.
+   */
+  @property({ type: String, reflect: true })
+  alt: string | null = null;
+
+  /**
+   * Specify the avatar disabled state.
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  @query("a,button")
+  private _focusableElement!: HTMLAnchorElement | HTMLButtonElement | null;
+
+  constructor() {
+    super();
+
+    this.addEventListener("click", (event: MouseEvent): void => {
+      if (
+        this.disabled ||
+        (this.as === "link" && this.href == null) ||
+        this.as === "icon"
```

Comment:
I don't think we should call `stopImmediatePropagation` when type is `icon`.

```suggestion
        (this.as === "link" && this.href == null)
```